File: solutions/dart/difference-of-squares/1/lib/difference_of_squares.dart
class DifferenceOfSquares {
  num squareOfSum(int n) {
    return (n * (n + 1) * (2 * n + 1)) / 6;
  }

  num sumOfSquares(int n) {
    return (n * (n + 1) * (2 * n + 1) * (2 * n + 2)) / 6;
  }

  num differenceOfSquares(int n) {
    return squareOfSum(n) - sumOfSquares(n);
  }
}

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    sum=0
    for i in range(1, number+1):
        sum+=i
    
    return sum*sum

def sum_of_squares(number):
    sum=0
    for i in range(1, number+1):
        sum+=i*i
    return sum

def difference_of_squares(number):
    return square_of_sum(number)-sum_of_squares(number)
